heka/plugins/s3splitfile/s3splitfile_input.go

```diff
@@ -10,19 +10,26 @@ import (
 	"fmt"
 	"github.com/crowdmob/goamz/aws"
 	"github.com/crowdmob/goamz/s3"
+	"github.com/mozilla-services/heka/message"
 	"github.com/mozilla-services/heka/pipeline"
 	"io"
 	"strings"
 	"sync"
+	"sync/atomic"
 	"time"
 )
 
 type S3SplitFileInput struct {
 	*S3SplitFileInputConfig
-	bucket   *s3.Bucket
-	schema   Schema
-	stop     chan bool
-	listChan chan string
+	bucket                 *s3.Bucket
+	schema                 Schema
+	stop                   chan bool
+	listChan               chan string
+	processFileCount       int64
```

Comment:
To avoid any memory alignment issues on 32 bit systems place these int64 values at the beginning of the structure